### ISSUE TITLE
[Proposal]Expose minWidth and minExtendedWidth from NavigationRail to NavbarRouter

### ISSUE THREAD

**Issue by bebaoboy** [maheshmnj/navbar_router#44]
Hello, it's me your badge buddy again 😂 This issue is not about a bug, I just wanna ask how to change the background, width, and selected icon color for the navigation rail?
In this screenshot, the navrail is a bit wide, and the selected icon color is... white (the one that appears to be disappeared)

![Screenshot_2024-04-19-23-22-30-397_com.iotecksolutions.todoapp-edit.jpg](https://github.com/maheshmnj/navbar_router/assets/74720131/49711f01-34f8-4c47-9874-54fabb6c4619)

Thanks!

**Comment by maheshmnj** [comment]
Thanks for filing the issue @bebaoboy,  this is currently not possible in navbar_router we need to expose two properties
```
minWidth and minExtendedWidth
```
 from NavigationRail to NavbarRouter 

<img width="489" alt="image" src="https://github.com/maheshmnj/navbar_router/assets/31410839/8c49fd10-6682-4783-9dd8-6c1bd4dffa7a">

If you are interested you are welcome to send a PR the change goes here

https://github.com/maheshmnj/navbar_router/blob/8f269d6df76b603c8bb0b9c267ff2807e0efb7ef/lib/src/animated_navbar.dart#L316-L318

**Comment by bebaoboy** [comment]
Thank you for replying.
And what about the selected icon color and background color for that navigation rail? 😅 I have no clue how to change them.

**Comment by maheshmnj** [comment]
> And what about the selected icon color and background color for that navigation rail? 😅 I have no clue how to change them.

Oops missed that part, For anykind of decoration to NavbarRouter you can use the decorationProperty the selectedicon is referred to as indicator to change the background shape and the icon color use indicatorShape and IndicatorColor property

```dart
 NavbarRouter(
      ...
      
        decoration: NavbarDecoration(
          indicatorColor: Colors.red,
          indicatorShape: ContinuousRectangleBorder(
              borderRadius: BorderRadius.circular(20)),
          height: 80,
          
          ...
          )
        )
          
```


<img width="265" alt="image" src="https://github.com/maheshmnj/navbar_router/assets/31410839/3063ebe9-c342-496b-9a40-7fb660e5286c">

**Comment by maheshmnj** [comment]
Apologies the docs are not well written: https://docs.maheshjamdade.com/navbar_router/types#decorating-the-navbar

@bebaoboy NavbarRouter has a decoration property which should allow you to customize the styling. It supports four different types of decoration corresponding to NavbarRouter.types

For selectedIcon color try using `IndicatorColor` and `indicatorShape` to change the shape, See [NavbarDecoration](https://github.com/maheshmnj/navbar_router/blob/main/lib/src/navbar_decoration.dar) class for all the available properties.

**Comment by maheshmnj** [comment]
The docs have been updated.

**Comment by maheshmnj** [comment]
@bebaoboy This is now fixed in https://github.com/maheshmnj/navbar_router/pull/47, These new properties will be available in upcoming release v0.7.4
